soc/nxp/mcx/mcxn/soc.c

```diff
@@ -44,6 +44,11 @@ DT_FOREACH_STATUS_OKAY(nxp_lpspi, FLEXCOMM_CHECK)
 /* This function is also called at deep sleep resume. */
 static int second_core_boot(void)
 {
+	/* Configure CPU1 TrustZone access level before CPU1 is enabled */
+	AHBSC->MASTER_SEC_LEVEL |=
+		AHBSC_MASTER_SEC_LEVEL_CPU1(CONFIG_SECOND_CORE_MCUX_ACCESS_LEVEL);
+	AHBSC->MASTER_SEC_ANTI_POL_REG = (~AHBSC->MASTER_SEC_LEVEL & 0x3FFFFFF3U) | 0x80000000U;
```

Comment:
Bits 3-2 (CPU1 field) are cleared by the mask `& 0x3FFFFFF3U`, which doesn't seem to be the intended behavior.

If COP1 in `MASTER_SEC_LEVEL` is set to 1 or 2 (_Non-secure and privileged Master_, respectively _Secure and non-privileged Master_), `MASTER_SEC_ANTI_POL_REG` will not have the inverse polarity of `MASTER_SEC_LEVEL` as specified.

IMHO, the mask should be set to `0x3FFFFFFFU`.

Reply:
Shouldn't macros be used instead of hard-coded values?

Concretely:
```C
AHBSC->MASTER_SEC_ANTI_POL_REG = (~AHBSC->MASTER_SEC_LEVEL & ~AHBSC_MASTER_SEC_ANTI_POL_REG_MASTER_SEC_LEVEL_ANTIPOL_LOCK_MASK) | AHBSC_MASTER_SEC_ANTI_POL_REG_MASTER_SEC_LEVEL_ANTIPOL_LOCK(2);
```

Reply:
Great feedback, @alxrey .  Updated as you requested.  Thanks for reviewing